File: contests/Codeforces Round #643 Div 2/Game_with_array.kt
// https://codeforces.com/contest/1355/problem/D
fun main() {
    var (n, s) = readLine()!!.split(' ').map {it.toInt()}
    
    if (s >= (2 * n)) {
        println("YES")
        val builder = StringBuilder()
        for (i in 0 .. (n - 2)) {
            builder.append("2 ")
        }
        builder.append((s - (2 * (n - 1))).toString())
        println(builder.toString())
        println(1)
    } else {
        println("NO")
    }
}

File: contests/Codeforces Round #643 Div 2/Game_with_array.cpp
// https://codeforces.com/contest/1355/problem/D
#include <iostream>
using namespace std;
int main() {
    int n, s;
    cin >> n >> s;
    if (s >= (n << 1)) {
        cout << "YES" << endl;
        string ans = "";
        for (int i = 0; i < (n - 1); ++i) {
            ans += "2 ";
        }
        cout << ans;
        cout << (s - 2 * (n - 1)) << endl;
        cout << 1 << endl;
    } else {
        cout << "NO" << endl;
    }
    return 0;
}